stingray/crossspectrum.py

```diff
@@ -0,0 +1,384 @@
+        return freqs[freqs > 0], cross
+
+    def rebin(self, df, method="mean"):
+        """
+        Rebin the cross spectrum to a new frequency resolution df.
+
+        Parameters
+        ----------
+        df: float
+            The new frequency resolution
+
+        Returns
+        -------
+        bin_cs = Crossspectrum object
+            The newly binned cross spectrum
+        """
+
+        # rebin cross spectrum to new resolution
+        binfreq, bincs, step_size = utils.rebin_data(self.freq[1:],
+                                                     self.cs[1:], df,
+                                                     method=method)
+
+        # make an empty cross spectrum object
+        bin_cs = Crossspectrum()
+
+        # store the binned periodogram in the new object
+        bin_cs.freq = np.hstack([binfreq[0] - self.df, binfreq])
+        bin_cs.cs = np.hstack([self.cs[0], bincs])
+        bin_cs.df = df
+        bin_cs.n = self.n
+        bin_cs.norm = self.norm
+        bin_cs.nphots_1 = self.nphots_1
+        bin_cs.nphots_2 = self.nphots_2
+        bin_cs.m = int(step_size)
+
+        return bin_cs
+
+    def _normalize_crossspectrum(self, unnorm_cs, tseg):
+        """
+        Normalize the real part of the cross spectrum to Leahy, absolute rms^2,
+        fractional rms^2 normalization, or not at all.
+
+        Parameters
+        ----------
+        unnorm_cs: numpy.ndarray
+            The unnormalized cross spectrum.
+
+        tseg: int
+            The length of the Fourier segment, in seconds.
+
+        Returns
+        -------
+        cs: numpy.nd.array
+            The normalized co-spectrum (real part of the cross spectrum). For
+            'none' normalization, imaginary part is returned as well.
+        """
+
+        # The "effective" count rate is the geometrical mean of the count rates
+        # of the two light curves; need to divide by tseg to have count rate
+        actual_mean = np.sqrt(self.nphots_1 * self.nphots_2 / tseg)
+
+        assert actual_mean > 0.0, \
+            "Mean count rate is <= 0. Something went wrong."
+
+        if self.norm.lower() == 'leahy':
+            c = unnorm_cs.real
+            cs = c * 2. / actual_mean
+
+        elif self.norm.lower() == 'frac':
+            c = unnorm_cs.real / np.float(self.n**2.)
+            cs = c * 2. * tseg / (actual_mean**2.0)
+
+        elif self.norm.lower() == 'abs':
+            c = unnorm_cs.real / np.float(self.n**2.)
+            cs = c * (2. * tseg)
+
+        elif self.norm.lower() == 'none':
+            cs = unnorm_cs
+
+        else:
+            raise Exception("Normalization not recognized!")
+
+        return cs
+
+    def coherence(self, lc_1, lc_2):
+
+        cross = self._fourier_cross(lc_1, lc_2)
+        crosslc_1 = self._fourier_cross(lc_1, lc_1)
+        crosslc_2 = self._fourier_cross(lc_2, lc_2)
+        coherence = (abs(cross)**2) / (crosslc_1 * crosslc_2)
+        return coherence
+
```

Comment:
Hello i tried coherence and made pep8 changes .please have a look . If you want me to try something please direct me  :)

PS: All commits are due to my intial commits to stingray(branch bispectra) , only " Added coherence " is a commit i made